config/config-server/resources/cruise-config.xsd

```diff
@@ -143,19 +143,58 @@
     <xsd:complexType name="configRepoType">
         <xsd:sequence>
             <xsd:choice minOccurs="1" maxOccurs="1">
-                <xsd:element name="svn" type="svnType"/>
-                <xsd:element name="hg" type="hgType"/>
-                <xsd:element name="p4" type="p4Type"/>
-                <xsd:element name="git" type="gitType"/>
-                <xsd:element name="tfs" type="tfsType"/>
-                <xsd:element name="scm" type="scmReferenceType"/>
+                <xsd:element name="svn" type="configRepoSvnType"/>
+                <xsd:element name="p4" type="configRepoP4Type"/>
+                <xsd:element name="git" type="configRepoGitType"/>
+                <xsd:element name="hg" type="configRepoHgType"/>
+                <xsd:element name="tfs" type="configRepoTfsType"/>
+                <xsd:element name="scm" type="configRepoScmReferenceType"/>
             </xsd:choice>
             <xsd:element name="configuration" minOccurs="0" maxOccurs="1" type="configurationType"/>
         </xsd:sequence>
         <xsd:attribute type="nameType" name="id" use="required"/>
         <xsd:attribute type="nameType" name="pluginId" use="optional"/>
     </xsd:complexType>
-
+    <xsd:complexType name="configRepoSvnType">
+        <xsd:attribute name="url" type="xsd:string" use="required"/>
+        <xsd:attribute name="username" type="xsd:string"/>
+        <xsd:attribute name="password" type="xsd:string"/>
+        <xsd:attribute name="checkexternals" type="xsd:boolean"/>
+        <xsd:attribute name="encryptedPassword" type="xsd:string"/>
+        <xsd:attribute name="materialName" type="nameType" use="optional"/>
+    </xsd:complexType>
+    <xsd:complexType name="configRepoP4Type">
+        <xsd:sequence>
+            <xsd:element name="view" minOccurs="1" maxOccurs="1"/>
+        </xsd:sequence>
+        <xsd:attribute name="port" type="xsd:string" use="required"/>
+        <xsd:attribute name="username" type="xsd:string"/>
+        <xsd:attribute name="password" type="xsd:string"/>
+        <xsd:attribute name="encryptedPassword" type="xsd:string"/>
+        <xsd:attribute name="useTickets" type="xsd:boolean" use="optional"/>
+        <xsd:attribute name="materialName" type="nameType" use="optional"/>
+    </xsd:complexType>
+    <xsd:complexType name="configRepoGitType">
+        <xsd:attribute name="url" type="xsd:string" use="required"/>
+        <xsd:attribute name="branch" type="xsd:string" use="optional"/>
```

Comment:
any specific reason for getting rid of `shallowClone`? I think it should be available.